test/grep.js

```diff
@@ -162,6 +180,30 @@ test('-i option', t => {
   t.is(result.split('\n').length - 1, 3);
 });
 
+test('one directory provided with -r option', t => {
+  const result = shell.grep('-r', 'test', 'test/resources');
+  t.falsy(shell.error());
+  t.is(result.split('\n').length, 29);
+});
+
+test('multiple directories or files provided', t => {
```

Comment:
We should probably split this into two separate test cases for clarity.

1. We want a test case where multiple directories/files are provided, but they do not overlap. ex. `"test/", "different-folder/file.txt"`
2. We want a test case (like this one) where there is overlap. We should assert that the overlapping files are double-counted (because that's what POSIX grep does). ex. `"test/", "test/file.txt"` (file.txt should be `grep`ed twice).